integration_tests/src/fixture_generator/generator.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under both the MIT license found in the
+// LICENSE-MIT file in the root directory of this source tree and the Apache
+// License, Version 2.0 found in the LICENSE-APACHE file in the root directory
+// of this source tree.
+
+//! This module contains the struct definitions of the tool output and main
+//! fixture generation logic.
+
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::Write;
+
+use akd::directory::Directory;
+use akd::storage::types::{AkdLabel, AkdValue, DbRecord};
+use akd::storage::StorageUtil;
+use clap::Parser;
+use rand::{rngs::OsRng, Rng};
+use serde::{Deserialize, Serialize};
+use winter_crypto::hashers::Blake3_256;
+use winter_math::fields::f128::BaseElement;
+
+use crate::fixture_generator::parser::Args;
+use crate::fixture_generator::writer::yaml::YamlWriter;
+use crate::fixture_generator::writer::Writer;
+
+type Blake3 = Blake3_256<BaseElement>;
+
+/// Directory state comprises all database records at a particular epoch.
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+pub struct State {
+    pub epoch: u32,
+    pub records: Vec<DbRecord>,
+}
+
+/// Delta comprises all updates published to the directory to advance to an
+/// epoch.
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+pub struct Delta {
+    pub epoch: u32,
+    pub updates: Vec<(AkdLabel, AkdValue)>,
+}
+
+/// Metadata about the output, including arguments passed to this tool and
+/// the tool version.
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+pub struct Metadata {
+    pub args: Args,
+    pub version: String,
+}
+
+// "@" has to be separated from "generated" or linters might ignore this file
```

Comment:
Why is that the case?

Reply:
Many tools (linters, CI) use the string "@​generated" as a marker for a non-human created file, and therefore ignore these files. If I write the "@​generated" string directly in `generator.rs`, `generator.rs` might be ignored even though it is a human-written file 🙂

Reply:
I thought that was the case when the file starts with `@generated`, not contains one. Thanks for the explanation!